Fig.Core/InvariantStringConverter.cs

```diff
@@ -1,19 +1,25 @@
 using System;
 using System.ComponentModel;
+using System.Linq;
 
 namespace Fig
 {
     internal class InvariantStringConverter : IStringConverter
     {
         public T Convert<T>(string value)
         {
-            return (T) Convert(value, typeof(T));
+            return (T)Convert(value, typeof(T));
         }
 
         public object Convert(string value, Type targetType)
         {
             var typeConverter = TypeDescriptor.GetConverter(targetType);
 
+            if (typeConverter is ArrayConverter && !typeConverter.CanConvertFrom(typeof(string)))
```

Comment:
Curious here, which types can be converted direct from comma-separated list of strings? Or in other words, what is the reason for the second part of the condition?

Reply:
The only possibility would be if a custom type descriptor was used for a custom object.